lib/board.rb

```diff
@@ -21,8 +21,68 @@ def initialize
                "D4" => Cell.new("D4")
               }
   end
+  # looping through - nested loop for expansion
 
   def valid_coordinate?(coordinate)
     cells.keys.include?(coordinate)
   end
+
+  def all_same?(array)
+    array.uniq.length == 1 
+  end
+
+  def not_all_same?(array)
+    array.uniq.length > 1 
+  end
+
+  def valid_placement?(ship, cells)
+    letters = cells.map do |cell| 
+      cell.split('').first
+    end
+    #should we convert the letters to integers here instead of at the end?
+
+    numbers = cells.map do |cell|
+      cell.split('').last.to_i
+    end
+    # Could this be a case expression? 
+    # TDD by the books?
+    if ship.length != cells.length
+      false 
+
+    elsif not_all_same?(letters) && not_all_same?(numbers)
```

Comment:
Should we make helper methods out of each of these conditional statements? Or is that excessive?

Reply:
I think this is readable enough. I'm not sure what the convention is on nesting helpers inside helpers....